pkg/controllers/disruption/orchestration/queue.go

```
@@ -0,0 +1,309 @@
+			cmd.LastError = err
+			q.RateLimitingInterface.AddRateLimited(cmd)
+			return reconcile.Result{RequeueAfter: controller.Immediately}, nil
+		}
+		// If the command failed, bail on the action.
+		// 1. Emit metrics for launch failures
+		// 2. Ensure cluster state no longer thinks these nodes are deleting
+		// 3. Remove it from the Queue's internal data structure
+		failedLaunches := lo.Filter(cmd.ReplacementKeys, func(key *NodeClaimKey, _ int) bool {
+			return !key.Initialized
+		})
+		disruptionReplacementNodeClaimFailedCounter.With(map[string]string{
+			methodLabel:            cmd.Method,
+			consolidationTypeLabel: cmd.ConsolidationType,
+		}).Add(float64(len(failedLaunches)))
+		q.cluster.UnmarkForDeletion(lo.Map(cmd.Candidates, func(s *state.StateNode, _ int) string { return s.ProviderID() })...)
+		multiErr := multierr.Combine(err, state.RequireNoScheduleTaint(ctx, q.kubeClient, false, cmd.Candidates...), cmd.LastError)
+
+		nodeNames := strings.Join(lo.Map(cmd.Candidates, func(s *state.StateNode, _ int) string {
+			return s.Name()
+		}), ",")
+		q.Remove(cmd)
+		logging.FromContext(ctx).With("nodeNames", nodeNames).Errorf("failed to disrupt nodes, %w", multiErr)
+		return reconcile.Result{RequeueAfter: controller.Immediately}, nil
+	}
+	// If command is complete, remove command from queue.
+	q.Remove(cmd)
+	return reconcile.Result{RequeueAfter: controller.Immediately}, nil
+}
+
+func (q *Queue) Process(ctx context.Context, cmd *Command) error {
+	if q.clock.Since(cmd.TimeAdded) > maxRetryDuration {
+		return NewUnrecoverableError(fmt.Errorf("command at %s reached timeout after %s", cmd.TimeAdded, q.clock.Since(cmd.TimeAdded)))
+	}
+	// If the time hasn't expired, either wait or terminate.
+	if err := q.WaitOrTerminate(ctx, cmd); err != nil {
+		// If there was an error, set this as the command's last error so that we can propagate it.
+		if IsUnrecoverableError(err) {
+			return fmt.Errorf("got unrecoverable error, %w", err)
```

Is wrapping it like this necessary or can we just return the unwrapped error?